iothub/device/src/IotHubBaseClient.cs

```diff
@@ -401,16 +401,28 @@ public async Task CloseAsync(CancellationToken cancellationToken = default)
             await InnerHandler.CloseAsync(cancellationToken).ConfigureAwait(false);
         }
 
-        /// <summary>
-        /// Releases the unmanaged resources used by the client and optionally disposes of the managed resources.
-        /// </summary>
-        /// <remarks>
-        /// The method <see cref="CloseAsync(CancellationToken)"/> should be called before disposing.
-        /// </remarks>
-        public void Dispose()
+        /// <inheritdoc/>
+        public async ValueTask DisposeAsync()
         {
-            Dispose(true);
-            GC.SuppressFinalize(this);
+            if (Logging.IsEnabled)
+                Logging.Enter(this, "Closing and disposing", nameof(DisposeAsync));
+
+            try
+            {
+                await CloseAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
This seems like an appropriate time to pass in a cancellation token that the SDK creates in order to avoid any sort of timeout error causing a deadlock. Maybe a minute at most?

Reply:
This pattern seems fine to me otherwise, though. (Signed off)